src/pages/index.js

```diff
@@ -16,42 +18,82 @@ import {
 const editFormElement = document.querySelector("#profile-edit-modal");
 const addFormElement = document.querySelector("#add-card-modal");
 const profileEditBtn = document.querySelector("#profile-edit-button");
+const avatarEditBtn = document.querySelector("#profile-image-button");
+const AvatarEditForm = document.forms["edit-avatar-form"];
+const avatar = document.querySelector(".profile__image");
 
 // Popups
 
 const imagePopup = new PopupWIthImage({
   popupSelector: "#preview-image-modal",
 });
 
-const editProfilePopup = new PopupWithForm({
-  popupSelector: "#profile-edit-modal",
-  handleFormSubmit: (data) => {
-    userInfo.setUserInfo(data);
-    editProfilePopup.close();
-    editFormValidator.disableSubmitButton();
-  },
-});
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo({
+      name: data.name,
+      about: data.about,
+      avatar: data.avatar,
+    });
+  })
+  .catch((err) => console.error("Error Fetching User Info", err));
 
 const userInfo = new UserInfo({
   nameSelector: "#profile-title",
   jobSelector: "#profile-description",
+  avatarSelector: ".profile__image",
+});
+
+const editProfilePopup = new PopupWithForm({
+  popupSelector: "#profile-edit-modal",
+  handleFormSubmit: (data) => {
+    editProfilePopup.setIsLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```